src/mlpack/tests/main_tests/linear_svm_test.cpp

```diff
@@ -0,0 +1,866 @@
+{
+  arma::mat trainData;
+  if (!data::Load("iris.csv", trainData))
+    BOOST_FAIL("Cannot load test dataset iris.csv!");
+
+  arma::Row<size_t> trainLabels;
+  if (!data::Load("iris_labels.txt", trainLabels))
+    BOOST_FAIL("Cannot load test dataset iris_labels.txt!");
+
+  SetInputParam("training", trainData);
+  SetInputParam("labels", trainLabels);
+
+  // First solution.
+  mlpackMain();
+
+  // Get the parameters of the output model obtained after first training.
+  const arma::mat parameters1 =
+      std::move(CLI::GetParam<LinearSVM<>*>("output_model")
+                ->Parameters());
+
+  // Reset the settings.
+  bindings::tests::CleanMemory();
+  CLI::ClearSettings();
+  CLI::RestoreSettings(testName);
+
+  SetInputParam("training", std::move(trainData));
+  SetInputParam("labels", std::move(trainLabels));
+  SetInputParam("no_intercept", bool(true));
+
+  // Second solution.
+  mlpackMain();
+
+  // Get the parameters of the output model obtained after second training.
+  const arma::mat& parameters2 =
+      CLI::GetParam<LinearSVM<>*>("output_model")->Parameters();
+
+  // Both solutions should be not equal.
+  CheckMatricesNotEqual(parameters1, parameters2);
+}
+
+/**
+ * The test is only compiled if the user has not specified OpenMP to be
+ * used and we test these case in one core.
+ */
+#ifndef HAS_OPENMP
+/**
+  * Ensuring that step size for optimizer is non negative.
+ **/
+BOOST_AUTO_TEST_CASE(LinearSVMNonNegativeStepSizeTest)
+{
+  arma::mat trainData;
+  if (!data::Load("iris.csv", trainData))
+    BOOST_FAIL("Cannot load test dataset iris.csv!");
+
+  arma::Row<size_t> trainLabels;
+  if (!data::Load("iris_labels.txt", trainLabels))
+    BOOST_FAIL("Cannot load test dataset iris_labels.txt!");
+
+  SetInputParam("training", std::move(trainData));
+  SetInputParam("labels", std::move(trainLabels));
+  SetInputParam("optimizer", std::string("psgd"));
+  SetInputParam("step_size", double(-0.01));
+
+  // Step size for optimizer is negative. It should throw a runtime error.
+  Log::Fatal.ignoreInput = true;
+  BOOST_REQUIRE_THROW(mlpackMain(), std::runtime_error);
+  Log::Fatal.ignoreInput = false;
+}
+
+/**
+  * Ensuring that epochs has some effects on the output.
+ **/
+BOOST_AUTO_TEST_CASE(LinearSVMDiffEpochsTest)
+{
+  arma::mat trainData = "2 0 0;"
+                        "0 0 0;"
+                        "0 2 1;"
+                        "1 0 2;"
+                        "0 1 0";
+
+  arma::Row<size_t> trainLabels = "1 0 1";
+
+  SetInputParam("training", trainData);
+  SetInputParam("labels", trainLabels);
+  SetInputParam("optimizer", std::string("psgd"));
+  SetInputParam("epochs", int(10));
+
+  // First solution.
+  mlpack::math::FixedRandomSeed();
+  mlpackMain();
+
+  // Get the parameters of the output model obtained after first training.
+  const arma::mat parameters1 =
+      std::move(CLI::GetParam<LinearSVM<>*>("output_model")
+                ->Parameters());
```

Comment:
This could probably look a little cleaner like this:

```
const arma::mat parameters1 = std::move(
    CLI::GetParam<LinearSVM<>*>("output_model")->Parameters();
```

I think that will fit in 80 characters... (not 100% sure though)

Reply:
Updated